core/src/main/java/org/apache/druid/java/util/http/client/HttpClientConfig.java

```diff
@@ -118,6 +121,11 @@ public int getNumConnections()
     return numConnections;
   }
 
+  public boolean isEagerInitialization()
+  {
+    return eagerInitialization;
```

Comment:
Nit: I feel `lazy` is a more commonly used term for such use cases rather than `eager`. I would prefer `lazyConnections` rather than `eagerInitialization`.

Reply:
done